### PR TITLE
Micro reddit

### PR DESCRIPTION
**Pull request by uryelah**
This is a clone of a simpler Reddit version only with it's models, associations, validations in order to simulate the database behaviour. What we did:

- create User model
- create Post model
- create Comment model
- add the relationships between the models
- add validations for the columns in the models

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments, dependent: :destroy
+  validates :user_id, presence: true
```
Comment (mosaaleb) [inline app/models/post.rb]: - [x] `belongs_to` association already adds presence validation to `user_id`. So, we don't need it.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  validates :user_id, presence: true
+  validates :post_id, presence: true
```
Comment (mosaaleb) [inline app/models/comment.rb]: - [x] Same for `user_id` and `post_id`

**Review by mosaaleb — APPROVED** [review]
### Well done @uryelah @jairjy :clap:

#### All requirements are done :100:

Please take a look at my comments below before moving on.

#### You can merge :+1:

###### Muhammad Ebeid

File: app/models/post.rb (hunk 1 above)
Comment (uryelah) [reply app/models/post.rb]: Thanks for the heads-up on that!

**Pull request merged**